devicetypes/smartenit/smartenit-metering-dual-load-controller.src/smartenit-metering-dual-load-controller.groovy

```diff
@@ -0,0 +1,177 @@
+/**
+ *  MLC30
+ *
+ *  Copyright 2021 Luis Contreras
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License. You may obtain a copy of the License at:
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software distributed under the License is distributed
+ *  on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License
+ *  for the specific language governing permissions and limitations under the License.
+ */
+ 
+ import groovy.transform.Field
+ import physicalgraph.zigbee.zcl.DataType
+ 
+ @Field final CurrentLevel = 0x0000
+ @Field final MoveToLevelWOnOff = 0x0004
+ @Field final MeteringCurrentSummation = 0x0000
+ @Field final MeteringInstantDemand = 0x0400
+ @Field final EnergyDivisor = 100000
+ @Field final CurrentDivisor = 100
+ @Field final Current = 0x00f0
+ @Field final Voltage = 0x00f1
+ @Field final OnOff = 0x0000
+ @Field final SmartenitMfrCode = 0x1075
+
+metadata {
+	definition (name: "Smartenit Metering Dual Load Controller", namespace: "Smartenit", author: "Luis Contreras", mnmn: "SmartThingsCommunity", vid: "472dac67-bbdd-344e-944b-43abafeeb82b") {
+		capability "Actuator"
+		capability "Configuration"
+		capability "Refresh"
+		capability "Power Meter"
+		capability "Energy Meter"
+		capability "Health Check"
+		capability "Voltage Measurement"
+		capability "monthpublic25501.current"
+		capability "monthpublic25501.load1"
+		capability "monthpublic25501.load2"
+		capability "monthpublic25501.levelControl"
+
+		fingerprint model: "ZBMLC30NC", manufacturer: "Smartenit, Inc", deviceJoinName: "Smartenit Switch"
+		fingerprint model: "ZBMLC30NO", manufacturer: "Smartenit, Inc", deviceJoinName: "Smartenit Switch"
+	}
+}
+
+def getFPoint(String FPointHex){
+	return (Float)Long.parseLong(FPointHex, 16)
+}
+
+// Parse incoming device messages to generate events
+def parse(String description) {
+	log.debug "Basic description: ${description}"
+	def event = zigbee.getEvent(description)
+	Map eventDescMap = zigbee.parseDescriptionAsMap(description)
+
+	if (description?.startsWith("on/off")) {
+		def cmds = zigbee.readAttribute(zigbee.ONOFF_CLUSTER, OnOff) + zigbee.readAttribute(zigbee.ONOFF_CLUSTER, OnOff, [destEndpoint: 2])
+		return cmds.collect { new physicalgraph.device.HubAction(it) }
+	}
+
+	if (event && event.name != "switch") {
+		log.debug "Collecting event: ${event}, ${event.name}, ${event.value}"
+		if ((eventDescMap?.sourceEndpoint == "01") || (eventDescMap?.endpoint == "01")) {
+			if (event.name == "power") {
+				return createEvent(name: "power", value: (event.value/EnergyDivisor))
+			} else {
+				sendEvent(event) 
```

Comment:
Perhaps you haven't known that, but in Groovy an object returned from last call of a method, is actually returned also by this whole method, for example.
```
def someMethod() {
    def object  = createEvent()
    object
}
```
(note that you don't have to use `return` statement, but it's not an error if you do, obviously)
and also
```
def someMethod() {
    createEvent()
}
```
would return created event in `someMethod`. 
Keeping that in mind, what I wrote in https://github.com/SmartThingsCommunity/SmartThingsPublic/pull/70915#discussion_r670229771 also applies here, as call `sendEvent(event)` would be the last in `parse()` method.
So if we want to keep things consistent, can you rework rest of the code, in a way that it would utilize `createEvent()`?

Reply:
Ah I wasn't aware of that! Thank you so much for the explanation, that was very helpful.